.github/workflows/ai-docs.yml

```diff
@@ -0,0 +1,45 @@
+name: Generate AI Documentation
+
+on:
+  pull_request:
```

Comment:
Would it be better to run this post merge? So it seamlessly adds the necessary documentation? I could see some cases where a conflict arises but it would be a bit more efficient imo.

Reply:
Good thinking! It would require a lot of `pull`s that we can avoid. Plus, the action already checks for changes prior committing, so having it run on based on specific PR file changes is unnecessary this way.